src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -865,50 +970,97 @@ private async Task<Dictionary<KeyValueIdentifier, ConfigurationSetting>> LoadKey
+                        loadOption.SnapshotName,
+                        cancellationToken);
                 }
-            }
 
-            return watchedSettings;
-        }
+                await CallWithRequestTracing(async () =>
+                {
+                    await foreach (Page<ConfigurationSetting> page in _options.PageableManager.GetPages(pageableSettings).ConfigureAwait(false))
+                    {
+                        foreach (ConfigurationSetting setting in page.Values)
+                        {
+                            keyValueChanges.Add(new KeyValueChange()
+                            {
+                                Key = setting.Key,
+                                Label = setting.Label,
+                                Current = setting,
+                                ChangeType = KeyValueChangeType.Modified,
+                                IsWatchedSetting = false
+                            });
+
+                            if (existingSettings.ContainsKey(setting.Key))
+                            {
+                                existingKeys.Add(setting.Key);
+                            }
+                        }
+                    }
+                }).ConfigureAwait(false);
+            }
 
-        private async Task<List<KeyValueChange>> GetRefreshedKeyValueCollections(
-            IEnumerable<KeyValueWatcher> multiKeyWatchers,
-            ConfigurationClient client,
-            StringBuilder logDebugBuilder,
-            StringBuilder logInfoBuilder,
-            Uri endpoint,
-            CancellationToken cancellationToken)
-        {
-            var keyValueChanges = new List<KeyValueChange>();
+            var loadedSettings = new HashSet<string>();
 
-            foreach (KeyValueWatcher changeWatcher in multiKeyWatchers)
+            foreach (KeyValueChange change in keyValueChanges)
             {
-                IEnumerable<ConfigurationSetting> currentKeyValues = GetCurrentKeyValueCollection(changeWatcher.Key, changeWatcher.Label, _watchedSettings.Values);
+                loadedSettings.Add(change.Key);
+            }
 
-                keyValueChanges.AddRange(
-                    await client.GetKeyValueChangeCollection(
-                        currentKeyValues,
-                        new GetKeyValueChangeCollectionOptions
-                        {
-                            KeyFilter = changeWatcher.Key,
-                            Label = changeWatcher.Label.NormalizeNull(),
-                            RequestTracingEnabled = _requestTracingEnabled,
-                            RequestTracingOptions = _requestTracingOptions
-                        },
-                        logDebugBuilder,
-                        logInfoBuilder,
-                        endpoint,
-                        cancellationToken)
-                    .ConfigureAwait(false));
+            foreach (string key in existingKeys)
```

Comment:
I may be missing something here, but it looks like the `exisitingKeys` are a subset of `keyValueChanges` and therefore a subset of `loadedSettings`. I can't see how we will ever enter the if-statement below.